crates/ruff_python_formatter/src/statement/stmt_try.rs

```diff
@@ -1,12 +1,150 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
+use crate::comments;
+use crate::comments::{
+    dangling_node_comments, leading_comments, leading_node_comments, trailing_node_comments,
+};
+use crate::prelude::*;
+use crate::statement::FormatRefWithRule;
+use crate::trivia::lines_before;
+use crate::{FormatNodeRule, PyFormatter};
 use ruff_formatter::{write, Buffer, FormatResult};
+use ruff_python_ast::node::AstNode;
+use ruff_python_ast::prelude::{ExceptHandler, ExceptHandlerExceptHandler, Ranged};
 use rustpython_parser::ast::StmtTry;
 
 #[derive(Default)]
 pub struct FormatStmtTry;
 
+#[derive(Copy, Clone, Default)]
+pub struct FormatExceptHandler;
+
+impl FormatRule<ExceptHandler, PyFormatContext<'_>> for FormatExceptHandler {
```

Comment:
i couldn't quite figure out why i need to implement `FormatRule` rather than just `FormatNodeRule` for the excepthandler

Reply:
`FormatNodeRule` requires an `AstNode`, but `ExceptHandler` is only a `Node`. @MichaReiser Do you why `ExceptHandler` is not `AstNode`.

Reply:
`ExceptHandler` is a union like `Stmt` and unions cannot implement `AstNodeRef` because it's impossible to convert a reference to a `Stmt` node (e.g. `&IfStmt`) to a statement reference (`&Stmt`) because all variants in `Stmt` take an *owned* value. 

That said, there's a `ExceptHandlerExceptHandler` formatter in `other/except_handler_except_handler.rs` (this name ....) 

We should move the formatting logic to the `except_handler_except_handler.rs` file and the union formatting here should simply dispatch to the item, the same as we do for `Stmt`

https://github.com/astral-sh/ruff/blob/66fcb55f27abda52425ce0138775c2fd9ee30959/crates/ruff_python_formatter/src/statement/mod.rs#L37-L69

Reply:
ah thanks. though how come it's a union? there only seems to be a single member

Reply:
RustPython uses Python's AST definition and it is defined as a union there. The same applies for TypeIgnore

```
  excepthandler = ExceptHandler(expr? type, identifier? name, stmt* body)
                    attributes (int lineno, int col_offset, int? end_lineno, int? end_col_offset)
```

No idea why. We're leaning towards tailoring our own AST optimized for our use case (e.g. introducing a `Body` node) at some point in the future.